index.js

```diff
@@ -316,6 +315,94 @@ const legacyScriptingRunner = (Zap, zcli, input) => {
     return _.template(templateString, options)(values);
   };
 
+  const ensureIsType = (result, type) => {
+    if (type) {
+      if (type.startsWith('array-')) {
+        if (Array.isArray(result)) {
+          return result;
+        } else if (result && typeof result === 'object') {
```

Comment:
I've left this one alone since Chang-Hung wrote it originally, I just moved it into a function. I've changed the instances that I added elsewhere to use lodash.